### ISSUE TITLE
FragmentCuller stops working with FragmentHighlighter

### ISSUE THREAD

**Issue by vulevukusej** [IFCjs/components#149]
### Describe the bug 📝

In my testing it appears that the fragmentCuller doesn't work for meshes created by the fragmentHighlighter.  In the example below, I've called the highlightById function on the FragmentIdMap associated with each storey in the model.  While this works, the performance takes quite a hit and I noticed that none of the objects were being culled when out of view.

After calling the function I set needsUpdate on the culler to true.  Any tips on how to circumvent this?

![image](https://github.com/IFCjs/components/assets/83825269/509767b5-ecb3-4888-a6b3-47faa8df321d)

### System Info 💻

```shell
version: 1.1.4
```

### Used Package Manager 📦

npm

### Validations ✅

- [X] Read the [docs](https://platform.thatopen.com/documentation).
- [X] Check that there isn't [already an issue](https://github.com/IFCjs/components/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a IFC.js components issue and not a framework-specific issue. For example, if it's a THREE.js related bug, it should likely be reported to [mrdoob/threejs](https://github.com/mrdoob/three.js) instead.
- [X] Check that this is a concrete bug. For Q&A join our [Community](https://people.thatopen.com/).
- [ ] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by vulevukusej** [comment]
Could this line be the issue: ? 
https://github.com/IFCjs/components/blob/54f63c4a0daee442110533c5f364cf3787219c37/src/fragments/FragmentHighlighter/index.ts#L427C15-L427C15

**Comment by vulevukusej** [comment]
Ok I figured out the solution:

I changed [this line](https://github.com/IFCjs/components/blob/54f63c4a0daee442110533c5f364cf3787219c37/src/fragments/FragmentHighlighter/index.ts#L427C15-L427C15) to true, and in the next line I added the new fragmentMesh to the culler. So far so good ;)

**Comment by RyugaRyuzaki** [comment]
> Could this line be the issue: ? https://github.com/IFCjs/components/blob/54f63c4a0daee442110533c5f364cf3787219c37/src/fragments/FragmentHighlighter/index.ts#L427C15-L427C15

hi, when you load model, did you call this https://github.com/IFCjs/components/blob/main/src/core/ScreenCuller/index.ts#L146?

**Comment by vulevukusej** [comment]
I believe the issue lies in the way the fragment highlighter works.  It creates a mesh for each object which isn't automatically added to the fragment culler.  I tried to modify this behaviour by adding the new meshes to the fragment culler and this actually solved the issue initially.  However, this introduced weird behaviour when deactivating the highlighter: some meshes were invisible afterwards.

**Comment by RyugaRyuzaki** [comment]
> I believe the issue lies in the way the fragment highlighter works. It creates a mesh for each object which isn't automatically added to the fragment culler. I tried to modify this behaviour by adding the new meshes to the fragment culler and this actually solved the issue initially. However, this introduced weird behaviour when deactivating the highlighter: some meshes were invisible afterwards.

I figured out some different. If i install via npm package it got some issue, however i download this file https://github.com/IFCjs/components/blob/main/resources/openbim-components.js , issue is gone. or event i use server side rendering is OK.

**Comment by agviegas** [comment]
Hum 🤔 I understand that you are using the FragmentHighlighter as a way to recolorize the meshes. If we provided a component to recolorize without using the highlighter, would this solve your use case @vulevukusej?

**Comment by vulevukusej** [comment]
@agviegas you read my mind. That's exactly what I want.  I think it'd be relevant for a lot of use cases, think scheduling, issues, version comparison, cost analysis etc.  Let me know if I can help as well 👌

**Comment by HoyosJuan** [comment]
Hey @vulevukusej! I did two things here:

1. I have added a configuration in the highlighter to decide if the highlight meshes are added to the culler or not. By default, is true. This was merged by PR https://github.com/IFCjs/components/pull/238.
2. Feature issue https://github.com/IFCjs/components/issues/226 already proposes to colorize fragment meshes just as you mentioned above. So, I will close this issue here and hopefully implement the feature asap.

Thanks!
